src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/SqlClientEventSource.cs

```diff
@@ -0,0 +1,344 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Text;
+using System.Diagnostics.Tracing;
+using System.Threading;
+
+namespace Microsoft.Data.SqlClient
+{
+    /// <include file='../../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlClientEventSource.xml' path='docs/members[@name="SqlClientEventSource"]/SqlClientEventSource/*'/>
+    [EventSource(Name = "Microsoft.Data.SqlClient.EventSource")]
+    public class SqlClientEventSource : EventSource
```

Comment:
Why are you exposing a new Type in the API surface area? We should just reference this event using the Name provided above i.e. `Microsoft.Data.SqlClient.EventSource` 
The reason I see that you are using this strong type `SqlClientEventSource` is while instantiating the listener to create the `EventSource` object. However EventSource object can also be created using simply the name of the event 
Ref Link https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.tracing.eventsource.-ctor?view=netframework-4.8#System_Diagnostics_Tracing_EventSource__ctor_System_String_

Reply:
Let's not expose the new type. It becomes a maintenance nightmare in future. Bid was awesome while it was, and now it is not. However the good thing is that there was no public type exposed by Bid, simply Guids and strings that we could reference. We should follow similar philosophy for event source mechanism as well.

Reply:
Done.